pkgs/development/python-modules/kurbopy/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, fonttools
+, pytestCheckHook
+, python
+, rustPlatform
+, unzip
+}:
+
+buildPythonPackage rec {
+  pname = "kurbopy";
+  version = "0.10.40";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-dhpcDi20Na6SDbRxrC8N3SWdN1J/CWJgCUI3scJX/6s=";
+  };
+
+  propagatedBuildInputs = [
+    fonttools
+  ];
+  nativeBuildInputs = [
+    rustPlatform.cargoSetupHook
+    rustPlatform.maturinBuildHook
+  ];
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    hash = "sha256-V3LeT0dqkfft1ftc+azwvuSzzdUJ7/wAp31fN7te9RQ=";
+  };
+
+  doCheck = true;
+  nativeCheckInputs = [
+    pytestCheckHook
+    unzip
+  ];
+  preCheck = ''
+    unzip -o $dist/*.whl
```

Comment:
Hmm interesting. Could you explain this a bit in a comment?

Reply:
This is a bit of a dodgy hack and I shouldn't have left it in (not without a comment).

The actual problem is the classic pytest import confusion. Because of pytest's manipulations of sys.path, the test process ends up importing `kurbopy/__init__.py` from the source directory, which does not contain the built .so, and so it fails like this:

```
/nix/store/qp5zys77biz7imbk6yy85q5pdv7qk84j-python3-3.11.6/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_basic.py:1: in <module>
    from kurbopy import Point, Vec2, TranslateScale
kurbopy/__init__.py:1: in <module>
    from .kurbopy import Point as Point
E   ModuleNotFoundError: No module named 'kurbopy.kurbopy'
```

We want it to import kurbopy from the nix store via PYTHONPATH instead.

So this hack was stealing `$dist` and using it to populate the source directory with a usable copy of the built .so, but a proper solution would be to make pytest import kurbopy from the installed directory like a proper installCheckPhase is supposed to do.

Pytest has an `--import-mode` option which should help with that kind of mess. But no matter what I do, `sys.path` always imports from `.` first. I think it's because `pytestCheckHook` does `python -m pytest`, and that's one of the side effects [noted in the docs](https://docs.pytest.org/en/7.1.x/how-to/usage.html#calling-pytest-through-python-m-pytest).

Anyway, I can just hack it to delete the module from the source directory in `preCheck`, so that it imports the installed copy from the nix store instead. With a comment explaining what's going on. :-)

Reply:
> Anyway, I can just hack it to delete the module from the source directory in `preCheck`, so that it imports the installed copy from the nix store instead. With a comment explaining what's going on. :-)

Thanks for the thorough explanation @danc86 . Could you do me please 1 more favor? In the very well written comment, could you mention that this is a common issue in nixpkgs and it is investigated in https://github.com/NixOS/nixpkgs/issues/255262 ?

Reply:
The PR should be ready after issue https://github.com/NixOS/nixpkgs/issues/255262 is mentioned.